File: fibf.c
#include <stdio.h>

long fib(long n) {
  long int a = 0;
  long int b = 1;
  for (long int i = 0; i <= n; i++) {
    long int c = a;
    a += b;
    b = c;
  }
  return a;
}

int main(void) {
  printf("%li", fib(46));
  return 0;
}

File: fibf.zig
const warn = @import("std").debug.warn;

fn fib(n: u32) u32 {
  var a: u32 = 0;
  var b: u32 = 1;
  var i: u32 = 0;
  while (i <= n) {
    var c: u32 = a;
    a += b;
    b = c;
    i += 1;
  }
  return a;
}

pub fn main() void {
  warn("fib(46) = {}", fib(46));
}
